django/db/models/query.py

```diff
@@ -519,6 +518,7 @@ def bulk_create(self, objs, batch_size=None, ignore_conflicts=False):
                 returned_columns = self._batched_insert(
                     objs_without_pk, fields, batch_size, ignore_conflicts=ignore_conflicts,
                 )
+                connection = connections[self.db]
```

Comment:
This would basically only improve things if you're passing in objects with a `pk` - which the lion's share of `bulk_create` usages probably _won't_ be doing.

Reply:
I think it's worth doing.